### PR TITLE
Add code snippets

### PR DESCRIPTION
**Pull request by maya**
This adds code snippets to the library site, displaying HTML markup of the components below each component.

### CODE REVIEW

File: _includes/scripts.html [hunk 1]
```diff
@@ -3,3 +3,4 @@
   <script src="{{ site.baseurl }}/assets/js/vendor/respond.js"></script>
   <script src="{{ site.baseurl }}/assets/js/vendor/rem.min.js"></script>
   <script src="{{ site.baseurl }}/assets/js/main.js"></script>
+  <script src="{{ site.baseurl }}/assets/js/vendor/prism.js"></script>
```
Comment (msecret) [inline _includes/scripts.html]: Vendor libs should go before our own, so this should be moved to the line before main.js

File: assets/js/main.js [hunk 2]
```diff
@@ -3,6 +3,36 @@ $(function(){
     $('.sidenav, .overlay').toggleClass('is-visible');
     e.preventDefault();
   });
+
+  var generateCodeSnippets = function(content, previewBox) {
+
+    var self = this;
+
+    this.parseCode = function(previewBox) {
```
Comment (msecret) [inline assets/js/main.js]: I don't understand the point of having theses functions as instance members on the generateCodeSnippets function if we're not using that as a class. Why wouldn't this just be:

```js
function parseCode(previewBox) {

}
```

Same goes for render.

File: assets/js/main.js [hunk 3]
```diff
@@ -3,6 +3,36 @@ $(function(){
...
+      var sampleCode = $('<div>');
+      $(sampleCode).html($(previewBox).html())
+      $(sampleCode).find('.is-peripheral').remove();
+      return sampleCode;
+    }
+
+    this.render = function(previewBox, sampleCode) {
+
+      var sampleCodeBox = $('<div class="usa-accordion usa-code-sample"><ul class="usa-unstyled-list"><li><button class="usa-button-unstyled" aria-expanded="false" aria-controls="collapsible-0">Toggle code</button><div id="collapsible-0" aria-hidden="true" class="usa-accordion-content"><pre><code class="language-markup"></code></pre></div></li></ul></div>');
```
Comment (msecret) [inline assets/js/main.js]: This long string is really hard to process. I'd suggest breaking it up onto newlines and using tabs to organize like actual html. Like so:

```js
var sampleCodeBox = $(''+
  '<div class="usa-accordion usa-code-sample">' +
    '<ul class="usa-unstyled-list">' +
      '<li><button ...etc
  
```

It's not perfect, but it'll improve the readability a little.

Comment (maya) [inline assets/js/main.js]: @msecret definitely, can u commit this to the branch?

File: _includes/scripts.html (hunk 1 above)
Comment (maya) [inline _includes/scripts.html]: Done in 7e25cbb.

File: assets/js/main.js (hunk 2 above)
Comment (maya) [inline assets/js/main.js]: @msecret This was to keep everything contained within the generateCodeSnippets function. Is this a problem?

Comment (msecret) [inline assets/js/main.js]: Declaring a function within generateCodeSnippets would contain it as well, so I'm still confused it's being declared on this rather the just being declared. Maybe to make unit testing easier? Maybe just add a TODO comment to rethink the structure of it.

File: assets/js/main.js [hunk 4]
```diff
@@ -3,6 +3,36 @@ $(function(){
...
+    this.parseCode = function(previewBox) {
+      var sampleCode = $('<div>');
+      $(sampleCode).html($(previewBox).html())
+      $(sampleCode).find('.is-peripheral').remove();
+      return sampleCode;
+    }
+
+    this.render = function(previewBox, sampleCode) {
+
+      var sampleCodeBox = $('<div class="usa-accordion-bordered usa-code-sample"><ul class="usa-unstyled-list"><li><button class="usa-button-unstyled" aria-expanded="false" aria-controls="collapsible-0">Code</button><div id="collapsible-0" aria-hidden="true" class="usa-accordion-content"><pre><code class="language-markup"></code></pre></div></li></ul></div>');
```
Comment (msecret) [inline assets/js/main.js]: I think my comment was lost, but could you clean up this huge string by breaking it up into lines and tabs?

**Comment by maya** [comment]
TODO: check why we are having these functions as instance members on the generateCodeSnippets function if we're not using that as a class. 

Possible reasons: 
- keep everything contained within the generateCodeSnippets function
- make it easier for unit testing

File: assets/js/main.js (hunk 3 above)
Comment (maya) [inline assets/js/main.js]: @msecret done in 5f55e89.

**Pull request merged**